src/openms/include/OpenMS/KERNEL/RangeManager.h

```diff
@@ -113,13 +113,13 @@ namespace OpenMS
     /// is @p value within [min, max]?
     bool contains(const double value) const
     {
-      return (min_ <= value) & (value <= max_);
+      return uint8_t(min_ <= value) & uint8_t(value <= max_); // using && leads to branches on all compilers in Debug and in Release on MVSC
```

Comment:
Note:
```
if ( (min_ <= value) && (value <= max_) ) return true; else false;
```
should AFAIK trigger the short circuit

Reply:
nope. With that solution: Clang at O1 still has branches, GCC at O0 also has branches. MSVC always branches.

Reply:
hmm strange... maybe it knows that branch prediction will always make the right choice anyway so it doesn't have performance impact?

Surprising... I thought in if (not a normal boolean expression) the second term is not allowed to be evaluated if the first one is false.